plasmapy/dispersion/numerical/hollweg_.py

```diff
@@ -0,0 +1,360 @@
+            )
+    if not (ion.is_ion or ion.is_category("element")):
+        raise ValueError("The particle passed for 'ion' must be an ion or element.")
+
+    # validate z_mean
+    if z_mean is None:
+        try:
+            z_mean = abs(ion.charge_number)
+        except ChargeError:
+            z_mean = 1
+    else:
+        if not isinstance(z_mean, (int, np.integer, float, np.floating)):
+            raise TypeError(
+                f"Expected int or float for argument 'z_mean', but got {type(z_mean)}."
+            )
+        z_mean = abs(z_mean)
+
+    # validate arguments
+    for arg_name in ("B", "n_i", "T_e", "T_i"):
+        val = locals()[arg_name].squeeze()
+        if val.shape != ():
+            raise ValueError(
+                f"Argument '{arg_name}' must be single valued and not an array of "
+                f"shape {val.shape}."
+            )
+        locals()[arg_name] = val
+
+    # validate arguments
+    for arg_name in ("gamma_e", "gamma_i"):
+        if not isinstance(locals()[arg_name], (int, np.integer, float, np.floating)):
+            raise TypeError(
+                f"Expected int or float for argument '{arg_name}', but got "
+                f"{type(locals()[arg_name])}."
+            )
+
+    # validate argument k
+    k = k.squeeze()
+    if not (k.ndim == 0 or k.ndim == 1):
+        raise ValueError(
+            f"Argument 'k' needs to be single valued or a 1D array astropy Quantity,"
+            f" got array of shape {k.shape}."
+        )
+    if np.any(k <= 0):
+        raise ValueError("Argument 'k' can not be a or have negative values.")
+
+    # validate argument theta
+    theta = theta.squeeze()
+    if not (theta.ndim == 0 or theta.ndim == 1):
+        raise ValueError(
+            f"Argument 'theta' needs to be a single valued or 1D array astropy "
+            f"Quantity, got array of shape {k.shape}."
+        )
+
+    # Single k value case
+    if np.isscalar(k.value):
+        k = np.array([k.value]) * u.rad / u.m
+
+    # Calc needed plasma parameters
+    with warnings.catch_warnings():
+        warnings.simplefilter("ignore", category=PhysicsWarning)
+        n_e = z_mean * n_i
+        c_s = pfp.ion_sound_speed(
+            T_e=T_e,
+            T_i=T_i,
+            ion=ion,
+            n_e=n_e,
+            gamma_e=gamma_e,
+            gamma_i=gamma_i,
+            z_mean=z_mean,
+        ).value
+        v_A = pfp.Alfven_speed(B, n_i, ion=ion, z_mean=z_mean).value
+        omega_ci = pfp.gyrofrequency(B=B, particle=ion, signed=False, Z=z_mean).value
+        omega_pe = pfp.plasma_frequency(n=n_e, particle="e-").value
+
+    # strip units from select input args
+    k = k.value
+    theta = theta.value
+    cs_vA = c_s / v_A
+
+    # Parameters kx and kz
+    kz = np.cos(theta) * k
+    kx = np.sin(theta) * k
+
+    # Define helpful parameters
+    beta = (c_s / v_A) ** 2
+    alpha_A = (k * v_A) ** 2
+    alpha_s = (k * c_s) ** 2  # == alpha_A * beta
```

Comment:
I was trading accuracy for performance here.  If we really wanted performance then we wouldn't be doing these readability declarations and just write everything out in one line.

The `alpha_A * beta` has a `v_A / v_A` and I just wanted to avoid that extra calculation.